python/tests/test_topology.py

```diff
@@ -4984,3 +4984,120 @@ def test_ts_single_tree_delete_middle(self):
         ts_keep = ts.delete_intervals([[0.25, 0.5]], record_provenance=False)
         ts_delete = ts.keep_intervals([[0, 0.25], [0.5, 1.0]], record_provenance=False)
         self.assertTrue(ts_equal(ts_keep, ts_delete))
+
+
+class TestTrim(unittest.TestCase):
+    """
+    Test the trimming functionality
+    """
+    def add_mutations(self, tables, position, ancestral_state, derived_states):
+        "Create a site at the specified position and assign mutations to arbitrary nodes"
+        site = tables.sites.add_row(position, ancestral_state)
+        for i, state in enumerate(derived_states):
+            tables.mutations.add_row(
+                site, i % tables.nodes.num_rows, state)
+
+    def test_failure_with_migrations(self):
+        ts = msprime.simulate(10, recombination_rate=2, random_seed=2)
+        ts = ts.keep_intervals([[0.1, 0.5]])
+        tables = ts.dump_tables()
+        tables.migrations.add_row(0, 1, 0, 0, 0, 0)
+        ts = tables.tree_sequence()
+        with self.assertRaises(ValueError):
```

Comment:
I don't think this does what you intend --- I think only the first one will be caught.

Reply:
Ah, you are right. I was being careless

Reply:
replaced with 

```
        self.assertRaises(ValueError, ts.ltrim)
        self.assertRaises(ValueError, ts.rtrim)
        self.assertRaises(ValueError, ts.trim)
```